server/src/main/kotlin/org/javacs/kt/overridemembers/OverrideMembers.kt

```diff
@@ -0,0 +1,282 @@
+
+fun listOverridableMembers(file: CompiledFile, cursor: Int): List<CodeAction> {
+    val kotlinClass = file.parseAtPoint(cursor)
+
+    if (kotlinClass is KtClass) {
+        return createOverrideAlternatives(file, kotlinClass)
+    }
+
+    return emptyList()
+}
+
+private fun createOverrideAlternatives(file: CompiledFile, kotlinClass: KtClass): List<CodeAction> {
+    // Get the functions that need to be implemented
+    val membersToImplement = getUnimplementedMembersStubs(file, kotlinClass)
+
+    val uri = file.parse.toPath().toUri().toString()
+
+    // Get the padding to be introduced before the member declarations
+    val padding = getDeclarationPadding(file, kotlinClass)
+
+    // Get the location where the new code will be placed
+    val newMembersStartPosition = getNewMembersStartPosition(file, kotlinClass)
+    
+    // loop through the memberstoimplement and create code actions
+    return membersToImplement.map { member ->
+        val newText = System.lineSeparator() + System.lineSeparator() + padding + member
+        val textEdit = TextEdit(Range(newMembersStartPosition, newMembersStartPosition), newText)
+
+        val codeAction = CodeAction()
+        codeAction.edit = WorkspaceEdit(mapOf(uri to listOf(textEdit)))
+        codeAction.title = member
+
+        codeAction
+    }
+}
+
+// TODO: any way can repeat less code between this and the getAbstractMembersStubs in the ImplementAbstractMembersQuickfix?
+private fun getUnimplementedMembersStubs(file: CompiledFile, kotlinClass: KtClass): List<String> =
+    // For each of the super types used by this class
+    // TODO: does not seem to handle the implicit Any and Object super types that well. Need to find out if that is easily solvable. Finds the methods from them if any super class or interface is present
+    kotlinClass
+        .superTypeListEntries
+        .mapNotNull {
+            // Find the definition of this super type
+            val referenceAtPoint = file.referenceExpressionAtPoint(it.startOffset)
+            val descriptor = referenceAtPoint?.second
+            val classDescriptor = getClassDescriptor(descriptor)
+
+            // If the super class is abstract, interface or just plain open
+            if (null != classDescriptor && classDescriptor.canBeExtended()) {
+                val superClassTypeArguments = getSuperClassTypeProjections(file, it)
+                classDescriptor
+                    .getMemberScope(superClassTypeArguments)
+                    .getContributedDescriptors()
+                    .filter { classMember ->
+                        (classMember is FunctionDescriptor &&
+                         classMember.canBeOverriden() &&
+                         !overridesDeclaration(kotlinClass, classMember)) ||
+                        (classMember is PropertyDescriptor &&
+                         classMember.canBeOverriden() &&
+                         !overridesDeclaration(kotlinClass, classMember))
+                    }
```

Comment:
I was going to propose factoring the `canBeOverriden` and `overridesDeclaration`, but then I saw that these are different overloads. Hmm. If we can use `MemberDescriptor`, we should be able to at least do

```suggestion
                    .filter { classMember ->
                        classMember.canBeOverriden() &&
                        (classMember is FunctionDescriptor && !overridesDeclaration(kotlinClass, classMember)) ||
                        (classMember is PropertyDescriptor && !overridesDeclaration(kotlinClass, classMember))
                    }
```

Reply:
Sadly no 🙁 Why? Because at this point `classMember` is sent in as a `DeclarationDescriptor` (`.getContributedDescriptors()` returns a `Collection<DeclarationDescriptor>`), so we would have to use another smart cast to make the above work.

Reply:
I think after the improvement in [596c6a3](https://github.com/fwcd/kotlin-language-server/pull/379/commits/596c6a3268e2fa6d9cbeb13daa551e03af464e54), there is not much more to be done. I think just casting to `MemberDescriptor` and calling the two methods in one go improves a lot 🙂